ObjectiveCloudantTests/QueryViewTests.m

```diff
@@ -0,0 +1,966 @@
+    XCTestExpectation *viewRowBlockCalled = [self expectationWithDescription:@"Document Found"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.ddoc = @"views101";
+    op.viewName = @"diet";
+    op.descending = YES;
+
+    __block int numberOfDocs = 0;
+    __block BOOL firstDoc = YES;
+    NSMutableArray<NSString *> *orderedKeys = [NSMutableArray array];
+
+    op.viewRowBlock = ^(NSDictionary<NSString *, NSObject *> *doc) {
+      if (firstDoc) {
+          firstDoc = NO;
+          [viewRowBlockCalled fulfill];
+      }
+      numberOfDocs++;
+      XCTAssertNotNil(doc);
+      XCTAssertNotNil(doc[@"id"]);
+      XCTAssertNotNil(doc[@"key"]);
+      XCTAssertNotNil(doc[@"value"]);
+
+      [orderedKeys addObject:doc[@"key"]];
+    };
+
+    op.queryViewCompletionBlock = ^(NSError *error) {
+      [requestCompleted fulfill];
+      XCTAssertNil(error);
+    };
+
+    [self.database addOperation:op];
+
+    [self waitForExpectationsWithTimeout:10.0f
+                                 handler:^(NSError *_Nullable error) {
+                                   NSLog(@"Operation completion handler not called");
+                                 }];
+
+    XCTAssertEqual(10, numberOfDocs);
+    XCTAssertEqualObjects(expectedOrderOfKeys, orderedKeys);
+}
+
+- (void)testfailsWithGroupingOnNonGroupableView
+{
+    XCTestExpectation *completionFailed = [self expectationWithDescription:@"Grouping on view"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.ddoc = @"views101";
+    op.viewName = @"diet";
+    op.group = YES;
+
+    op.queryViewCompletionBlock = ^(NSError *error) {
+      [completionFailed fulfill];
+      XCTAssertNotNil(error);
+    };
+
+    [self.database addOperation:op];
+
+    [self waitForExpectationsWithTimeout:10.0f
+                                 handler:^(NSError *_Nullable error) {
+                                   NSLog(@"Operation completion handler not called");
+                                 }];
+}
+
+- (void)testCorrectlyQueryViewInculdingDocs
+{
+    XCTestExpectation *requestCompleted =
+        [self expectationWithDescription:@"Succesfully view query"];
+    XCTestExpectation *viewRowBlockCalled = [self expectationWithDescription:@"Document Found"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.ddoc = @"views101";
+    op.viewName = @"diet";
+    op.includeDocs = YES;
+
+    __block int numberOfDocs = 0;
+    __block BOOL firstDoc = YES;
+
+    op.viewRowBlock = ^(NSDictionary<NSString *, NSObject *> *doc) {
+      if (firstDoc) {
+          firstDoc = NO;
+          [viewRowBlockCalled fulfill];
+      }
+      numberOfDocs++;
+      XCTAssertNotNil(doc);
+      XCTAssertNotNil(doc[@"id"]);
+      XCTAssertNotNil(doc[@"key"]);
+      XCTAssertNotNil(doc[@"value"]);
+      // Make sure the doc field is present.
+      XCTAssertNotNil(doc[@"doc"]);
```

Comment:
Seeing this test case makes me think we should perhaps assert that the `doc` field isn't present in the other tests where `includeDocs` is `NO` so we show we're not always including the `doc` field.